packages/app/src/app/components/Preview/DevTools/Console/index.js

```diff
@@ -37,31 +37,50 @@ class Console extends React.Component {
   handleMessage = data => {
     switch (data.type) {
       case 'console': {
-        const { method, args: jsonArgs } = data;
-        const args = CircularJSON.parse(jsonArgs);
-        this.addMessage(method, args);
+        const message = Decode(data.log);
+        const { method, data: args } = message;
+
+        switch (method) {
+          case 'clear': {
+            // If the event was done by the packager
+            const hideMessage = args[0] === 'internal';
+
+            this.clearConsole(hideMessage);
+            break;
+          }
+          default: {
+            this.addMessage(method, args);
+            break;
+          }
+        }
         break;
       }
       case 'clear-console': {
-        this.clearConsole();
+        if (this.state.initialClear) {
+          this.setState({
+            initialClear: false
+          })
+        } else {
+          this.clearConsole();
+        }
         break;
       }
       case 'eval-result': {
         const { result, error } = data;
 
-        const parsedJson = result ? CircularJSON.parse(result) : result;
+        const decoded = Decode(result);
 
         if (!error) {
-          this.addMessage('log', [parsedJson], 'return');
+          this.addMessage('return', [decoded]);
```

Comment:
Currently the `return` & `command` types are unimplemented in `console-feed` so it defaults to a standard log

Reply:
re-implemented :white_check_mark: